libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1200,7 +1203,15 @@ bool ExpressionCompiler::visit(MemberAccess const& _memberAccess)
 	// Check whether the member is a bound function.
 	ASTString const& member = _memberAccess.memberName();
 	if (auto funType = dynamic_cast<FunctionType const*>(_memberAccess.annotation().type))
-		if (funType->bound())
+	{
+		if (funType->kind() == FunctionType::Kind::Definition && member == "selector")
```

Comment:
Would it work/be better to rather `assert(member == "selector")` inside the if?

Reply:
Actually I think this whole change should have been removed... the correct case is the one further below in this function - this here is actually wrong and never reached, I think. I'll check and remove it, if true.

Reply:
Sounds good

Reply:
Yep, should have been removed, removed it now.